modules/ROOT/pages/for-each-scope-concept.adoc

```diff
@@ -4,182 +4,316 @@ include::_attributes.adoc[]
++
+[source,log]
 ----
-include::{examplesdir}/for-each-scope-concept_3.xml[]
+[Romaguera-Crona, Deckow-Crist, Romaguera-Jacobson, Robel-Corkery,
+ Keebler LLC, Considine-Lockman, Johns Group, Abernathy Group,
+ Yost and Sons, Hoeger LLC]
 ----
+. Logger 4 is located outside For Each. The expression `payload..company.name` in this logger produces the same array produced by Logger 3, but Logger 4 is able to select `payload` directly. For Each consumes the `rootMessage` variable, so the expression `vars.rootMessage.payload..company.name` from Logger 4 logger would return `null`.
+. Logger 5 is located outside For Each. The expression `attributes.statusCode` produces the same status code value as the expression in Logger 2, but Logger 5 is able to select `attributes` directly. For Each consumes the `rootMessage` variable, so the expression `vars.rootMessage.attributes.statusCode` from Logger 5 would return `null`.
 
-The `batchSize` attribute creates sub-arrays of two elements. So inside `foreach`, the payload will have an array of two elements.
-
-=== Aggregate the Process of Each Element into a Variable
+The next example illustrates the batch size (`batchsize`) functionality. Assume that the flow collects its input payload from the preceding flow through a Flow Reference (`<flow ref />`) component:
 
-[source,xml,linenums]
+[source,xml]
 ----
-include::{examplesdir}/for-each-scope-concept_4.xml[]
+<flow name="example-for-each-flow" >
+  ...
+  <!-- 1: Flow Reference sends Mule event to next flow. -->
+  <flow-ref doc:name="Flow Reference" name="for-each-batchExample"/>
+</flow>
+<flow name="for-each-batchExample" >
+  <!-- 2: For Each >
+  <foreach doc:name="For Each"
+               collection="#[payload]"
+               batchSize="3">
+    <!-- 3: Logger >
+    <logger level="INFO" doc:name="Logger"
+            message="#[payload.id]"/>
+  </foreach>
+</flow>
 ----
 
-Within the For Each, the example creates an `aggregation` variable that concatenates the name in reverse. Then that variable can be accessed outside the For Each scope.
+. The Flow Reference component sends its Mule event to next flow (`for-each-batchExample`). This event contains a JSON array of objects as its `payload`. This `payload` originated from the HTTP request in the previous flow (`example-for-each-flow`).
+. For Each (`<foreach />`) in this flow (`for-each-batchExample`) sets the batch size `batchSize="3"`.
+. The Logger in this flow prints the `id` values of the input (`payload.id`)  in batches of 3. Each batch is an array of up to 3 elements. Because there are only 10 elements from the split input to For Each, the Logger prints the following:
```

Comment:
```suggestion
. The Logger in this flow prints the `id` values of the input (`payload.id`) in batches of 3. Each batch is an array of up to 3 elements. Because there are only 10 elements from the split input to For Each, the Logger prints the following:
```